xarray/tests/test_dataset.py

```diff
@@ -1748,6 +1748,23 @@ def test_reindex(self):
         actual = ds.reindex(x=[0, 1, 3], y=[0, 1])
         assert_identical(expected, actual)
 
+    @requires_sparse
+    def test_reindex_sparse(self):
+        data = create_test_data()
+        dim3 = list("abdeghijk")
+        actual = data.reindex(dim3=dim3, sparse=True)
+        expected = data.reindex(dim3=dim3, sparse=False)
+        for k, v in data.data_vars.items():
+            np.testing.assert_equal(actual[k].data.todense(), expected[k].data)
+        assert actual['var3'].data.density < 1.0
+
+        data["var3"] = data["var3"].astype(int)
+        actual = data.reindex(dim3=dim3, sparse=True, fill_value=-10)
+        expected = data.reindex(dim3=dim3, sparse=False, fill_value=-10)
+        for k, v in data.data_vars.items():
+            np.testing.assert_equal(actual[k].data.todense(), expected[k].data)
```

Comment:
I think we generally use `assert_array_equal` for numpy arrays (but I can't immediately recall the difference...)

Reply:
I think these actually end up doing the exact same checks.

Reply:
`values` property does not work for the sparse-backed Variable, resulting in the failure of `assert_array_equal`.
I'll add TODO comment for this.